### PR TITLE
Implement SEO

### LINKED ISSUE #2: Add SEO Link Previews to the Website
Screenshot 2023-04-04 at 5 26 56 AM

Similar to the how the ACM main website shows link previews for each route dynamically with meta tags following the OpenGraph protocol, we can use the same code in this file to generate link previews on the Membership Portal.

https://github.com/acmucsd/main-website/blob/main/src/components/SEO/index.tsx

Since the majority of pages require authentication and can't properly be given custom previews for their route, we can implement a simpler approach with a single static link preview across the whole portal.

### PR DESCRIPTION
**Pull request by raymosun**
# Info

Closes #2 

# Description

What changes did you make? List all distinct problems that this PR addresses. Explain any relevant
motivation or context.

Adds SEO component with open graph and other metadata for link previews, etc.

## Changes

Added SEO component, used in page layout, to insert meta tags to page head.

# Type of Change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [ ] locally on Desktop.
- [ ] locally on mobile - use https://ngrok.io to get a copy on a mobile device
- [ ] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have run and passed all new and existing Cypress tests. Add screenshots below.

# Checklist

- [ ] I have performed a self-review of my own code.
- [ ] I have followed the style guidelines of this project.
- [ ] I have documented my code's `src/lib` functions and commented hard to understand areas
      anywhere else.
- [ ] My changes produce no new warnings.

# Screenshots

Please include a screenshot of your Cypress testing suite passing successfully.

If you made any visual changes to the website, please include relevant screenshots below.

image

### CODE REVIEW

**Comment by raymosun** [comment]
Initial progress on SEO component. Some descriptions, titles, etc. are placeholders.

File: src/components/layout/PageLayout/index.tsx [hunk 1]
```diff
@@ -9,6 +10,7 @@ interface LayoutProps {
 
 const PageLayout = ({ user, children }: PropsWithChildren<LayoutProps>) => (
   <>
+    <SEO />
```
Comment (farisashai) [inline src/components/layout/PageLayout/index.tsx]: Let's move this component to the `src/pages/_app.tsx` file for all technical components that aren't directly related to the UI layout.

File: src/components/common/SEO/index.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import Head from 'next/head';
+
+const SEO = () => {
+  const TITLE = 'ACM UCSD Membership Portal';
+  const DESC =
```
Comment (farisashai) [inline src/components/common/SEO/index.tsx:6]: Updated description (open to suggestions)
`Meet new friends, discover exciting events, and earn points for free ACM swag! Create an account today to find your community!`

File: src/components/common/SEO/index.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      {/* type of content */}
+      <meta property="og:type" content="website" />
+      {/* actual website title */}
+      <meta property="og:site_name" content="ACM at UCSD" />
+      {/* title to display for the specific link being shared */}
+      <meta property="og:title" content={TITLE} />
+      {/* preview image */}
+      <meta
+        property="og:image"
+        content="https://cdn.discordapp.com/attachments/975657067031437332/1092620423281250335/Screenshot_2023-04-03_at_6.23.24_PM.jpg"
```
Comment (farisashai) [inline src/components/common/SEO/index.tsx]: Screenshot 2023-04-14 at 12 51 26 PM

Let's keep this image just as the standard light mode ACM logo to match the style of the ACM main website link previews so its clean and legible.

**Review by farisashai — CHANGES_REQUESTED** [review]
Last comment is unrelated to the changes in your actual code but we should be prioritizing the performance, accessibility, and search engine optimizations on our website whenever we develop. We can only really test the login page in this project since the rest are authenticated but if you go to this website (https://pagespeed.web.dev/), it gives some good tips for general optimizations usually.

It will give us a score on each metric and some tips to improve them as you scroll below. When I compare the current version to your deployment preview, it looks like we bumped up the SEO successfully with no pending issues so good job with that 👏 

Screenshot 2023-04-14 at 12 59 40 PM

Screenshot 2023-04-14 at 1 00 53 PM


Using the deployment preview link in this PR, can you test it and see if you can address some of the accessibility issues we have right now?

**Review by farisashai — APPROVED** [review]
LGTM  🤑 🚀

**Pull request merged**